### PR TITLE
feat(angular-sample): add main angular sample app

### PR DESCRIPTION
**Pull request by daveadams56**
# JIRA Ticket

No JIRA ticket for this change; this is adding the full version of the Angular sample developed earlier this year, to develop.

# Description

Add the "full" version of the Angular sample to develop. This contains the functionality of the "complete" version linked to from the blog post, plus the registration page and SDK integration point comments. Documentation is covered by the blog post which is already live.

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [x] Documentation story is created and tracked.
- [x] UI is completed or ticket is created.
- [x] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [x] Acceptance criteria met
- [x] Spell-check run
- [x] Peer reviewed
- [x] Proofread

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,15 +38,14 @@ test-results
 samples/_static/js/*.js*
 samples/_static/js/*
 samples/*/public/app.*
-samples/*/todos
 samples/*/users
-todos/
-users/
+!samples/angular-todo/*/todos
```
Comment (cerebrl) [inline .gitignore:42]: Huh, that's odd. I thought the below `todos/` would only ignore a directory created at the root. Now I'm wondering how [the `todos` directory in the React sample](https://github.com/ForgeRock/forgerock-javascript-sdk/tree/develop/samples/reactjs-todo/client/components/todos) survived.

File: .gitignore [hunk 2]
```diff
@@ -38,15 +38,14 @@ test-results
...
 # Environment files
 .env
 .env.local
 .angular
 samples/**/.env.local
 samples/**/.env
-samples/angularjs-todo/.env.local
\ No newline at end of file
+todos/*
+users/*
```
Comment (cerebrl) [inline .gitignore:51]: Non-blocking: you might want to configure your IDE to always add as blank line at the end of files.

File: README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ To try out the ForgeRock JavaScript SDK please look at one of our samples:
 
 - [**React Todo - `/samples/reactjs-todo`**](./samples/reactjs-todo/README.md)
 
-Since this sample app uses npm's workspaces feature, npm 7 or higher is required. When on npm 7+, dependencies are already installed when running `npm install`. You can run this sample app with: `npm run start:reactjs-todo`.
+- [**Angular Todo - `/samples/angular-todo`**](./samples/angular-todo/README.md)
```
Comment (cerebrl) [inline README.md:61]: I think you accidentally replaced a paragraph with your addition. Can you add the note about needing npm 7 back?

File: package.json [hunk 4]
```diff
@@ -18,6 +18,7 @@
     "start:e2e": "nx run autoscript-apps:serve",
     "start:embedded-login": "nx serve embedded-login",
     "start:central-login": "nx serve central-login",
+    "start:angular-todo": "nx run angular-todo:config && nx run-many --target=serve --projects=todo-api,angular-todo",
```
Comment (cerebrl) [inline package.json:21]: Other than this change here, I think the rest of changes below to this file need to be reverted. These were changes recently introduced by @ryanbas21.

**Review by cerebrl — CHANGES_REQUESTED** [review]
The change to `package.json` and the root README need to be updated before merge. The `gitignore` is odd to me, but no change is required.

**Comment by cerebrl** [comment]
Oh, and please squash the commits into one and make sure the commit message and PR title use the correct syntax.

File: .gitignore (hunk 1 above)
Comment (daveadams56) [reply .gitignore:42]: Me neither!

Comment (daveadams56) [reply .gitignore:42]: For sanity let's add the same exception for the react app

File: README.md (hunk 3 above)
Comment (daveadams56) [reply README.md:61]: Ryan asked me to remove this paragraph

File: package.json (hunk 4 above)
Comment (daveadams56) [reply package.json:21]: Just fixed this with Ryan

**Comment by daveadams56** [comment]
> Oh, and please squash the commits into one and make sure the commit message and PR title use the correct syntax.

I will squash the commits today. Could you please confirm what you mean by correct syntax for the commit message and PR title?

File: .gitignore (hunk 2 above)
Comment (daveadams56) [reply .gitignore:51]: Will do, thanks for the suggestion

File: README.md (hunk 3 above)
Comment (daveadams56) [reply README.md:61]: Thinking about it, we don't specify in the main README the need to use npm 7. Perhaps we should have it in the **Requirements** section along with the Node version?

Comment (daveadams56) [reply README.md:65]: After discussing with @cerebrl I've now changed the main README to include npm 7 as a requirement

File: package.json (hunk 4 above)
Comment (daveadams56) [reply package.json:21]: Should be in line with develop now

File: package.json [hunk 5]
```diff
@@ -44,15 +45,17 @@
...
     "@nrwl/eslint-plugin-nx": "13.9.6",
     "@nrwl/express": "13.9.6",
     "@nrwl/jest": "13.9.6",
     "@nrwl/js": "13.9.6",
     "@nrwl/linter": "13.9.6",
     "@nrwl/node": "13.9.6",
-    "@nrwl/nx-cloud": "13.2.2",
+    "@nrwl/nx-cloud": "13.1.6",
     "@nrwl/react": "13.9.6",
+    "@nrwl/tao": "13.9.6",
```
Comment (cerebrl) [inline package.json:58]: I believe this too was intended to be removed in the same PR as the `pre-commit` PR referenced earlier. Unless you have an explicit needs to keep this in, I'd leave it out.

**Review by cerebrl — CHANGES_REQUESTED** [review]
I think there's still some unintended changes to the `package.json` that needs to be reviewed. Once these are fixed, I think it's ready.

File: package.json [hunk 6]
```diff
@@ -18,6 +18,7 @@
     "start:e2e": "nx run autoscript-apps:serve",
     "start:embedded-login": "nx serve embedded-login",
     "start:central-login": "nx serve central-login",
+    "start:angular-todo": "nx run angular-todo:config && nx run-many --target=serve --projects=todo-api,angular-todo",
     "start:reactjs-todo": "nx run-many --target=serve --projects=reactjs-todo, todo-api",
```
Comment (cerebrl) [inline package.json:22]: Unfortunately, there's a bug hidden in here. The space just before `todo-api` shouldn't be there. I have a PR open for it here: https://github.com/ForgeRock/forgerock-javascript-sdk/pull/196.

Comment (cerebrl) [reply package.json:22]: The fix PR was merged. Sorry to do this to you, but you'll need to rebase your PR again.

File: package.json [hunk 7]
```diff
@@ -30,7 +31,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "shx rm -rf dist/ && npm run format:staged && npm run lint && nx affected:build",
+      "pre-commit": "shx rm -rf dist/ && npm run format:staged && npm run lint && npm run build",
```
[the remark this answers is not in the document]
Comment (daveadams56) [reply package.json:34]: Ok I'm pretty sure I've fixed this now

File: package.json [hunk 8]
```diff
@@ -44,15 +45,17 @@
...
+    "@nrwl/angular": "^13.9.6",
     "@nrwl/cli": "13.9.6",
     "@nrwl/eslint-plugin-nx": "13.9.6",
     "@nrwl/express": "13.9.6",
     "@nrwl/jest": "13.9.6",
     "@nrwl/js": "13.9.6",
     "@nrwl/linter": "13.9.6",
     "@nrwl/node": "13.9.6",
-    "@nrwl/nx-cloud": "13.2.2",
+    "@nrwl/nx-cloud": "13.1.6",
```
[the remark this answers is not in the document]
Comment (daveadams56) [reply package.json:56]: Yes

File: package.json (hunk 5 above)
Comment (daveadams56) [reply package.json:58]: No this should have been removed. Fixed I think

**Review by cerebrl — APPROVED** [review]
Pulled in locally and everything ran well. Approved!

**Pull request merged**